tests/src/e2e-smoke.spec.ts

```diff
@@ -71,38 +84,38 @@ describe('Basic e2e verification of podman desktop start', async () => {
 
       await page.screenshot({ path: 'tests/output/screenshots/screenshot-welcome-page-init.png', fullPage: true });
 
-      const welcomeMessage = page.locator('text=/Welcome to Podman Desktop.*/');
-      await playExpect(welcomeMessage).toBeVisible();
+      const welcomePage = new WelcomePage(page);
+      await playExpect(welcomePage.welcomeMessage).toBeVisible();
     });
 
     test('Telemetry checkbox is present, set to true, consent can be changed', async () => {
       // wait for the initial screen to be loaded
-      const telemetryConsent = page.getByText('Telemetry');
-      expect(telemetryConsent).not.undefined;
-      expect(await telemetryConsent.isChecked()).to.be.true;
+      const welcomePage = new WelcomePage(page);
+      await playExpect(welcomePage.telemetryConsent).toBeVisible();
+      playExpect(await welcomePage.telemetryConsent.isChecked()).toBeTruthy();
 
-      await telemetryConsent.click();
-      expect(await telemetryConsent.isChecked()).to.be.false;
+      await welcomePage.turnOffTelemetry();
+      playExpect(await welcomePage.telemetryConsent.isChecked()).toBeFalsy();
     });
 
     test('Redirection from Welcome page to Dashboard works', async () => {
-      const goToPodmanDesktopButton = page.locator('button:text("Go to Podman Desktop")');
+      const welcomePage = new WelcomePage(page);
```

Comment:
I'm wondering if we should not delegate here as well to the runner

instead of 
```ts
const welcomePage = new WelcomePage(page);
```

we do
```ts
const welcomePage = runner.getPage(pages.Welcome);
```

like for the Kubernetes library

```ts
const k8sApi = kc.makeApiClient(k8s.CoreV1Api);
```

so we know all helpers we have for pages with completion

Reply:
this can be postponed (not a required change)

Reply:
Created an issues #3298